src/diffusers/schedulers/scheduling_ddpm.py

```diff
@@ -215,15 +215,18 @@ def _get_variance(self, t, predicted_variance=None, variance_type=None):
         # x_{t-1} ~ N(pred_prev_sample, variance) == add variance to pred_sample
         variance = (1 - alpha_prod_t_prev) / (1 - alpha_prod_t) * current_beta_t
 
+        # we always take the log of variance, so clamp it to ensure it's not 0
+        variance = torch.clamp(variance, min=1e-20)
+
         if variance_type is None:
             variance_type = self.config.variance_type
 
         # hacks - were probably added for training stability
         if variance_type == "fixed_small":
-            variance = torch.clamp(variance, min=1e-20)
+            variance = variance
```

Comment:
```suggestion
pass
```

Or remove this `if` condition.